### PR TITLE
DIT 674 - Extended Use Flag

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+          }
+        )
+      }
+    )
+
+  }
+
+  private def statusHasChanged(c: Case, status: Boolean): Boolean = {
+    !c.decision.map(_.applicationForExtendedUse).contains(status)
```
Comment (pllaurado) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala]: contains?

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala]: I'm not sure what your question is @pllaurado

Comment (pllaurado) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala]: I'm just curious  why you are comparing two booleans with contains. Not wrong but just estrange.  Maybe I would expect something like a filter and a isdefined.

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          reference,
+          c => successful(views.html.change_extended_use_status(c, errors))
+        )
+      },
+      (status: Boolean) => {
+        getCaseAndRespond(
+          reference,
+          c => {
+            if(statusHasChanged(c, status)) {
+              caseService.updateExtendedUseStatus(c, status, request.operator).flatMap { c =>
```
Comment (pllaurado) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala]: The redirection is the same even the status has not change. In this case I would move the status change logic/check to the service.

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala (hunk 1 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ExtendedUseCaseController.scala]: It's the same thing really

**Pull request merged**